app/Http/Controllers/Backend/StatisticController.php

```diff
@@ -110,4 +111,34 @@ public static function getWeeklyTravelTimeByUser(User $user, Carbon $since, Carb
                      return $row;
                  });
     }
+
+    public static function getTravelPurposes(User $user, Carbon $since, Carbon $until): Collection {
+        if ($since->isAfter($until)) {
+            throw new InvalidArgumentException('since cannot be after until');
+        }
+
+        return DB::table('train_checkins')
+                 ->join('statuses', 'train_checkins.status_id', '=', 'statuses.id')
+                 ->where('statuses.user_id', '=', $user->id)
+                 ->where('train_checkins.departure', '>=', $since->toIso8601String())
+                 ->where('train_checkins.departure', '<=', $until->toIso8601String())
+                 ->groupBy('statuses.business')
+                 ->select([
+                              DB::raw('statuses.business AS reason'),
+                              DB::raw('COUNT(*) AS count'),
+                              DB::raw('SUM(TIMESTAMPDIFF(MINUTE, departure, arrival)) AS duration')
+                          ])
+                 ->orderByDesc('duration')
+                 ->get()
+                 ->map(function($row) {
+                     if ($row->reason == Business::PRIVATE) {
+                         $row->reason = __('stationboard.business.private');
```

Comment:
That's kinda dirty. The backend should not spit out translated strings. Please change this to the number corresponding to our Enum and handle the translation in the frontend.